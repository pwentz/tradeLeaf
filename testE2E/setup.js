import React, { Component } from 'react';
const { View, StyleSheet } = require('react-native');

import { Provider } from 'react-redux';

import MockApi from './mocks/MockApi';
const { setupApp } = require('../setupApp');

__DEV__ = true;

jest.mock('Linking', () => {
  return {
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    openURL: jest.genMockFn().mockReturnValue(Promise.resolve()),
    canOpenURL: jest.genMockFn().mockReturnValue(Promise.resolve()),
    getInitialURL: jest.genMockFn().mockReturnValue(Promise.resolve()),
  };
});

jest.mock('ScrollView', () => jest.genMockFromModule('ScrollView'));

export const renderer = require('react-test-renderer');

const AccountRequirements = require('../src/components/register/AccountRequirements').default;
const App = require('../src/components/app/App').default;
const Avatar = require('../src/components/common/Avatar').default;
const Card = require('../src/components/matchboard/Card').default;
const LoginForm = require('../src/components/login/LoginForm').default;
const RegisterForm = require('../src/components/register/RegisterForm').default;
const TabBar = require('../src/components/app/TabBar').default;
const ChatPreview = require('../src/components/chat/ChatPreview').default;

export const components = {
  LoginForm,
  RegisterForm,
  Card,
  AccountRequirements,
  Avatar,
  TabBar,
  ChatPreview,
};

class TestRouteObserver {}

export function setupTestApp() {
  const api = new MockApi();

  const shouldLog = false;

  const { actions, store } = setupApp(api, shouldLog);

  const styles = StyleSheet.create({
    container: {
      flex: 1,
      justifyContent: 'center',
      alignItems: 'stretch',
      backgroundColor: 'white',
    },
  });

  const testRouteObserver = new TestRouteObserver();

  class TradeLeafApp extends Component {
    render() {
      return (
        <View style={styles.container}>
          <Provider store={store}>
            <App screenProps={{ actions, testRouteObserver }} />
          </Provider>
        </View>
      );
    }
  }

  return {
    TradeLeafApp,
    api,
    actions,
    dispatch: store.dispatch,
    testRouteObserver,
  };
}
