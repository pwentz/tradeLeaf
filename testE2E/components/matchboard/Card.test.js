import React from 'react';

import { components, renderer } from '../../setup';

const noop = () => {}

it('<Card />', () => {
  const Card = components.Card;
  const component = renderer.create(
    <Card />
  );
  expect(component.toJSON()).toMatchSnapshot();
});
