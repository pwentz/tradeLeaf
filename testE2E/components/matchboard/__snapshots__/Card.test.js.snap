// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "height": 20,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F5F5F5",
        "borderColor": "#B8B8B8",
        "borderRadius": 10,
        "borderWidth": 1,
        "flexDirection": "column",
        "height": 933.8,
        "justifyContent": "flex-end",
        "width": 637.5,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": "82%",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "borderRadius": 45,
              "height": 90,
              "width": 90,
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "uri": "https://inthecloud.png",
            }
          }
          style={
            Array [
              Object {
                "borderColor": "#B8B8B8",
                "borderWidth": 1,
              },
              Object {
                "borderRadius": 45,
                "height": 90,
                "width": 90,
              },
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 0.85,
            "flexDirection": "column",
            "justifyContent": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          @harry-jim
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Simple Logo
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          5
           miles away
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "height": "58.5%",
          "justifyContent": "flex-end",
          "marginBottom": 10,
          "marginTop": 10,
          "width": "75%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flexDirection": "column",
            "height": "66%",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          offer:
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          i hate cheese. please have
        </Text>
        <View
          style={
            Object {
              "height": "45%",
              "marginBottom": 5,
              "overflow": "hidden",
              "width": "100%",
            }
          }
        >
          <View
            onLayout={[Function]}
            style={undefined}
          >
            <View
              collapsable={undefined}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    undefined,
                  ]
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <Image
                  blurRadius={10}
                  onLoadEnd={[Function]}
                  source={
                    Object {
                      "uri": "https://imagethings.jpeg",
                    }
                  }
                  style={
                    Object {
                      "alignSelf": "center",
                      "aspectRatio": 2,
                      "width": "100%",
                    }
                  }
                />
              </View>
            </View>
            <Modal
              hardwareAccelerated={false}
              onRequestClose={[Function]}
              transparent={true}
              visible={false}
            >
              <View
                collapsable={undefined}
                style={
                  Object {
                    "backgroundColor": "black",
                    "height": 1334,
                    "left": 0,
                    "opacity": 0,
                    "position": "absolute",
                    "top": 0,
                    "width": 750,
                  }
                }
              />
              <View
                collapsable={undefined}
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "flex": 1,
                    "height": 0,
                    "justifyContent": "center",
                    "left": 0,
                    "position": "absolute",
                    "top": 0,
                    "width": 0,
                  }
                }
              >
                <Image
                  blurRadius={10}
                  onLoadEnd={[Function]}
                  source={
                    Object {
                      "uri": "https://imagethings.jpeg",
                    }
                  }
                  style={
                    Object {
                      "alignSelf": "center",
                      "aspectRatio": 2,
                      "width": "100%",
                    }
                  }
                />
              </View>
              <View
                collapsable={undefined}
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "left": 0,
                    "opacity": 0,
                    "position": "absolute",
                    "top": 0,
                    "width": 750,
                  }
                }
              >
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  collapsable={undefined}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "white",
                        "fontSize": 35,
                        "lineHeight": 40,
                        "shadowColor": "black",
                        "shadowOffset": Object {
                          "height": 0,
                          "width": 0,
                        },
                        "shadowOpacity": 0.8,
                        "shadowRadius": 1.5,
                        "textAlign": "center",
                        "width": 40,
                      }
                    }
                  >
                    Ã—
                  </Text>
                </View>
              </View>
            </Modal>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "borderTopColor": "#B8B8B8",
            "borderTopWidth": 1,
            "flexDirection": "column",
            "height": "34%",
            "justifyContent": "space-between",
            "paddingBottom": 5,
            "paddingTop": 5,
            "width": "100%",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          need:
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          bouquet of petunias
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderTopColor": "#B8B8B8",
          "borderTopWidth": 1,
          "flexDirection": "row",
          "height": "14.3%",
          "justifyContent": "space-between",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "flexDirection": "column",
            "marginLeft": 20,
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#B8B8B8",
                  "fontSize": 38,
                },
                undefined,
                Object {
                  "fontFamily": "Octicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            ï…¦
          </Text>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderColor": "#DFBA00",
            "borderRadius": 100,
            "borderWidth": 4,
            "height": 42,
            "opacity": 1,
            "width": 42,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      />
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "flex": 1,
            "flexDirection": "column",
            "marginRight": 20,
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#27496B",
                  "fontSize": 38,
                },
                undefined,
                Object {
                  "fontFamily": "Octicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            ï†¯
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
